src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -81,6 +81,13 @@ private void AddWellKnownTypes(IRootingServiceProvider rootProvider)
             {
                 rootProvider.AddCompilationRoot(stringType, "String type is always generated");
             }
+
+            var coreLib = _typeSystemContext.GetModuleForSimpleName("System.Private.Corelib");
+
+            // Adding for CPP codeGen
+            var eeTypeType = coreLib.GetKnownType("Internal.Runtime", "EEType");
+            rootProvider.AddCompilationRoot(eeTypeType, "Enables C++ Code Generation");
```

Comment:
Is this used for anything? I would expect CppCodegen just treating the EEType pointer as a `void*`. The pointer acquires a type when the classlib code handles it.